internal/controller/processor.go

```diff
@@ -123,48 +138,60 @@ func (processor *AppConfigurationProviderProcessor) processKeyValueRefresh(exist
 	// Check if the sentinel based refresh is enabled
 	if provider.Spec.Configuration.Refresh != nil &&
 		provider.Spec.Configuration.Refresh.Enabled {
-		processor.RefreshOptions.sentinelBasedRefreshEnabled = true
+		processor.RefreshOptions.keyValueRefreshEnabled = true
 	} else {
-		reconcileState.NextSentinelBasedRefreshReconcileTime = metav1.Time{}
+		reconcileState.NextKeyValueRefreshReconcileTime = metav1.Time{}
 		return nil
 	}
 
 	refreshInterval, _ := time.ParseDuration(provider.Spec.Configuration.Refresh.Interval)
-	nextSentinelBasedRefreshReconcileTime := metav1.Time{Time: processor.CurrentTime.Add(refreshInterval)}
+	nextKeyValueRefreshReconcileTime := metav1.Time{Time: processor.CurrentTime.Add(refreshInterval)}
 	if processor.ShouldReconcile {
-		reconcileState.NextSentinelBasedRefreshReconcileTime = nextSentinelBasedRefreshReconcileTime
+		reconcileState.NextKeyValueRefreshReconcileTime = nextKeyValueRefreshReconcileTime
 		return nil
 	}
 
-	if !processor.CurrentTime.After(reconcileState.NextSentinelBasedRefreshReconcileTime.Time) {
+	if !processor.CurrentTime.After(reconcileState.NextKeyValueRefreshReconcileTime.Time) {
 		return nil
 	}
 
-	if processor.RefreshOptions.sentinelChanged, processor.RefreshOptions.updatedSentinelETags, err = (*processor.Retriever).CheckAndRefreshSentinels(processor.Context, processor.Provider, reconcileState.SentinelETags); err != nil {
-		return err
-	}
-
-	if !processor.RefreshOptions.sentinelChanged {
-		reconcileState.NextSentinelBasedRefreshReconcileTime = nextSentinelBasedRefreshReconcileTime
-		return nil
-	}
 	// Get the latest key value settings
 	existingConfigMapSettings := &existingConfigMap.Data
 	if processor.Settings.ConfigMapSettings != nil {
 		existingConfigMapSettings = &processor.Settings.ConfigMapSettings
 	}
+
+	if provider.Spec.Configuration.Refresh.Monitoring != nil {
```

Comment:
Could we put the line164- line178 right before the line 158, seems don't need to read the existing data if etag checking decides not refresh